src/index.ts

```diff
@@ -0,0 +1,45 @@
+import { v4 as uuidv4 } from 'uuid';
+
+type AlgorithmFunction = () => string;
+
+const algorithms: Record<string, AlgorithmFunction> = {
+  uuid4: uuidv4,
+};
+
+function replacingAlgorithm(
+  memoizationTable: Record<string, string>,
+  _: unknown,
+  uriBase: string,
+  uuidAlgorithm: string,
+  originalUuid: string,
+) {
+  if (memoizationTable[`${uriBase}${originalUuid}`]) {
```

Comment:
Since the memoization doesn't include the algorithm, if 2 uris have the same uuid, but different algorithm, they will actually only use the algorithm from whichever was first in the document.

Reply:
Mmmm, I actually don't know what to do here, if 2 uris have the same uuid and different algorithm I think it's an error, we could silently fail and do the best we can that in this case is use the same algo, create another uuid (that I don't think is a good solution) or throw an error